crates/rpc-replay/src/block_context.rs

```diff
@@ -49,16 +56,21 @@ pub fn build_block_context(
         use_kzg_da,
     };
 
+    let strk_fee_token_address = get_env_var_or_default(
+        "SNOS_STRK_FEE_TOKEN_ADDRESS",
+        "0x04718f5a0fc34cc1af16a1cdee98ffb20c31f5cd61d6ab07201858f4287c938d",
+    );
+    let eth_fee_token_address = get_env_var_or_default(
+        "SNOS_ETH_FEE_TOKEN_ADDRESS",
+        "0x049d36570d4e46f48e99674bd3fcc84644ddd6b96f7c741b1562b82f9e004dc7",
+    );
+
     let chain_info = ChainInfo {
         chain_id,
         // cf. https://docs.starknet.io/tools/important-addresses/
         fee_token_addresses: FeeTokenAddresses {
-            strk_fee_token_address: contract_address!(
-                "0x04718f5a0fc34cc1af16a1cdee98ffb20c31f5cd61d6ab07201858f4287c938d"
-            ),
-            eth_fee_token_address: contract_address!(
-                "0x049d36570d4e46f48e99674bd3fcc84644ddd6b96f7c741b1562b82f9e004dc7"
-            ),
+            strk_fee_token_address: contract_address!(strk_fee_token_address.as_str()),
```

Comment:
Do we need `.as_str()` here? Adding a reference means `&String` will deref to `&str`
```suggestion
            strk_fee_token_address: contract_address!(&strk_fee_token_address),
```

Reply:
```rust
contract_address!(&eth_fee_token_address)
```
```
the trait bound `FeltConverter: TryIntoFelt<&String>` is not satisfied
the following other types implement trait `TryIntoFelt<V>`:
  `FeltConverter` implements `TryIntoFelt<&str>`
  `FeltConverter` implements `TryIntoFelt<u128>`
  `FeltConverter` implements `TryIntoFelt<u16>`
  `FeltConverter` implements `TryIntoFelt<u32>`
  `FeltConverter` implements `TryIntoFelt<u64>`
  `FeltConverter` implements `TryIntoFelt<u8>`
```

IMO yes we need this, can you think of any other approach ?

Reply:
You can keep it as it is